### PR TITLE
Upgrade Party - Play 2.7 & Compile-Time DI

### PR DESCRIPTION
**Pull request by mbarton**
I tried to move to t3 instance types but it was broken in `CODE` - failing to launch with an error about `akka-slf4j`. Probably related to #23. I've fixed it by upgrading ALL THE THINGS.

- Upgrade all the libraries yadda yadda.
- Switch to compile-time DI
- Get it running locally in DEV by hard-coding default properties and changing the port to not conflict with login.gutools.
- Remove the cloudformation in favour of a single dev bucket. 
- Use `media-service` profile credentials instead of static IAM credentials
- Rename the `Application` controller to `Api` as Play already has a class called `Application`

### CODE REVIEW

File: app/controllers/Api.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package controllers
```
Comment (akash1810) [inline app/controllers/Api.scala]: why change from `Application` to `Api`?

File: app/lib/Config.scala [hunk 2]
```diff
@@ -2,17 +2,26 @@ package lib
...
-  val properties = Properties.fromPath("/etc/gu/s3-uploader.properties")
+  val properties: Map[String, String] = Try(Properties.fromPath("/etc/gu/s3-uploader.properties")).getOrElse(Map.empty)
+
+  val awsCredentials = new AWSCredentialsProviderChain(
+    new ProfileCredentialsProvider("media-service"),
+    InstanceProfileCredentialsProvider.getInstance()
+  )
 
-  val awsCredentials = new DefaultAWSCredentialsProviderChain()
+  val region = properties.getOrElse("aws.region", "eu-west-1")
```
Comment (akash1810) [inline app/lib/Config.scala]: Presumably this means its zero-config in DEV?

File: app/controllers/Api.scala (hunk 1 above)
Comment (mbarton) [reply app/controllers/Api.scala]: It was clashing with `Application` in the Play API. I can try to put it back

File: app/lib/Config.scala (hunk 2 above)
Comment (mbarton) [reply app/lib/Config.scala]: Yeah it felt like the most straight-forward thing to do

**Pull request merged**